src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts

```diff
@@ -0,0 +1,228 @@
+
+  private emitters = new Map<string, VulnerabilityCountEmitter>();
+
+  constructor(
+    private readonly workspace: IVSCodeWorkspace,
+    private readonly window: IVSCodeWindow,
+    private readonly config: IConfiguration,
+    private readonly vulnerabilityCountProvider: ModuleVulnerabilityCountProvider,
+    private readonly ossService: OssService,
+    private readonly logger: ILog,
+  ) {}
+
+  activate(): boolean {
+    if (this.config.isDevelopment) {
+      // Do not activate service in production until we have feature flag story.
+      return false;
+    }
+
+    this.disposables.push(
+      this.workspace.onDidChangeTextDocument(ev => {
+        if (ev.contentChanges.length) {
+          this.processFile(ev.document);
+        }
+      }),
+      this.window.onDidChangeActiveTextEditor(ev => {
+        if (ev) {
+          this.processFile(ev.document);
+        }
+      }),
+    );
+
+    const activeEditor = this.window.getActiveTextEditor();
+    if (activeEditor) {
+      this.processFile(activeEditor.document);
+    }
+
+    return true;
+  }
+
+  dispose(): void {
+    while (this.disposables.length) {
+      const disposable = this.disposables.pop();
+      if (disposable) {
+        disposable.dispose();
+      }
+    }
+
+    for (const emitter of this.emitters.values()) {
+      emitter.removeAllListeners();
+    }
+  }
+
+  processFile(document: TextDocument): boolean {
+    if (!document) {
+      return false;
+    }
+
+    const { fileName, languageId } = document;
+    const supportedLanguage = this.getSupportedLanguage(fileName, languageId);
+    if (supportedLanguage === null || !this.shouldProcessFile(fileName, supportedLanguage)) {
+      return false;
+    }
+
+    let emitter = this.emitters.get(fileName);
+    if (emitter) {
+      emitter.removeAllListeners();
+    } else {
+      emitter = new VulnerabilityCountEmitter();
+      this.emitters.set(fileName, emitter);
+    }
+
+    // Todo: implement in package.json parsing in ROAD-294
+    // emitter.on('package', createPackageWatcher);
+
+    emitter.on(VulnerabilityCountEvents.Error, e => this.logger.error(`Error counting module vulnerabilities: ${e}`));
+
+    emitter.on(VulnerabilityCountEvents.Start, _packages => {
+      // Todo: ROAD-475
+      // flushDecorations(fileName, packages);
+    });
+    emitter.on(VulnerabilityCountEvents.Tested, _moduleInfo => {
+      // Todo: ROAD-475
+      // calculated(packageInfo);
+    });
+
+    emitter.on(VulnerabilityCountEvents.Done, _modules => {
+      // Todo: ROAD-475
+      // flushDecorations(fileName, packages, true);
+      // refreshDiagnostics(document, diagnostics, packages);
+    });
+
+    // Start
+    void this.getImportedModules(fileName, document.getText(), supportedLanguage, emitter);
+    return true;
+  }
+
+  private shouldProcessFile(fileName: string, language: SupportedLanguage): boolean {
+    if ([SupportedLanguage.TypeScript, SupportedLanguage.JavaScript].includes(language)) {
```

Comment:
Why only these two and not the other file types in the enum above?

Reply:
This is to prevent redundant parsing of JS/TS files by Babel if OSS results are not ready. In HTML case we'll query API directly, not relying on OSS result.

I'll add `PJSON` enum there in ROAD-294. Left it out for now because I'll need to think through some of the logic around it.